Wordle.Api/Wordle.Api/Services/LeaderboardService.cs

```diff
@@ -0,0 +1,74 @@
+﻿using Microsoft.EntityFrameworkCore;
+using System.Reflection.Metadata.Ecma335;
+using Wordle.Api.Models;
+using static System.Runtime.InteropServices.JavaScript.JSType;
+
+namespace Wordle.Api.Services;
+
+public class LeaderboardService
+{
+    //private readonly List<string> words = LoadWordList();
+    private static object _lock = new();
+
+    public WordleDbContext Db { get; set; }
+
+    public LeaderboardService(WordleDbContext db)
+    {
+        Db = db;
+    }
+
+    public async Task<List<Score>> GetTopTenScores()
+    {
+        /*for (int j = 0; j < players.Count; j++)
+        {
+            scores[j] = new string[3];
+        }
+
+        for (int i = 0; i < players.Count && i < 10; i++)
+        {
+            scores[i][0] = players[i].Name;
+            scores[i][1] = "" + players[i].AverageAttempts;
+            scores[i][2] = "" + players[i].GameCount;
+        }*/
+
+        return await Db.Players.
+            OrderBy(player => player.AverageAttempts).
+            Take(10).
+            Select(player => new Score
+            {
+                Name = player.Name,
+                AverageAttempts = player.AverageAttempts,
+                GameCount = player.GameCount
+            }).ToListAsync();
+    }
+
+    public async Task<bool> UpdateScore(NewScore score)
```

Comment:
```suggestion
    public async Task<bool> UpdateScore(Score score)
```

If you take suggestion of changing NewScore into Score